test/augmentation/test_augmentation.py

```diff
@@ -24,6 +25,794 @@
+        assert output.shape[0] == batch_shape[0]
+        assert transformation.shape == expected_transformation_shape
+
+        output, final_transformation = augmentation((test_input, test_transform), params=generated_params)
+        assert output.shape[0] == batch_shape[0]
+        assert final_transformation.shape == expected_transformation_shape
+        assert_allclose(final_transformation, transformation @ test_transform)
+
+        output, final_transformation = augmentation(
+            (test_input, test_transform), params=generated_params, return_transform=True)
+        assert output.shape[0] == batch_shape[0]
+        assert final_transformation.shape == expected_transformation_shape
+        assert_allclose(final_transformation, transformation @ test_transform)
+
+    def _test_cardinality_implementation(self, input_shape, expected_output_shape, params):
+
+        # p==0.0
+        augmentation = self._create_augmentation_from_params(**params, p=0.0)
+        test_input = torch.rand(input_shape, device=self.device, dtype=self.dtype)
+        output = augmentation(test_input)
+        assert len(output.shape) == 4
+        assert output.shape == torch.Size((1,) * (4 - len(input_shape)) + tuple(input_shape))
+
+        # p==1.0
+        augmentation = self._create_augmentation_from_params(**params, p=1.0)
+        test_input = torch.rand(input_shape, device=self.device, dtype=self.dtype)
+        output = augmentation(test_input)
+        assert len(output.shape) == 4
+        assert output.shape == expected_output_shape
+
+    def _test_random_p_0_implementation(self, params):
+        augmentation = self._create_augmentation_from_params(**params, p=0.0, return_transform=False)
+        expected_output_shape = torch.Size((2, 3, 4, 5))
+        test_input = torch.rand((2, 3, 4, 5), device=self.device, dtype=self.dtype)
+        output = augmentation(test_input)
+        assert (output == test_input).all()
+
+    def _test_random_p_0_return_transform_implementation(self, params):
+        augmentation = self._create_augmentation_from_params(**params, p=0.0, return_transform=True)
+        expected_output_shape = torch.Size((2, 3, 4, 5))
+        expected_transformation_shape = torch.Size((2, 3, 3))
+        test_input = torch.rand((2, 3, 4, 5), device=self.device, dtype=self.dtype)
+        output, transformation = augmentation(test_input)
+
+        assert (output == test_input).all()
+        assert transformation.shape == expected_transformation_shape
+        assert (transformation == kornia.eye_like(3, transformation)).all()
+
+    def _test_random_p_1_implementation(self, input_tensor, expected_output, params):
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=False)
+        output = augmentation(input_tensor.to(self.device).to(self.dtype))
+
+        # Output should match
+        assert output.shape == expected_output.shape
```

Comment:
no against, but might be redundant since it's the same as we test in cardinality, right ?